test/cloudnative/agentversion_set.go

```diff
@@ -0,0 +1,183 @@
+//go:build e2e
+
+package cloudnative
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/arch"
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/src/version"
+	"github.com/Dynatrace/dynatrace-operator/test/csi"
+	"github.com/Dynatrace/dynatrace-operator/test/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/test/kubeobjects/manifests"
+	"github.com/Dynatrace/dynatrace-operator/test/kubeobjects/namespace"
+	"github.com/Dynatrace/dynatrace-operator/test/sampleapps"
+	"github.com/Dynatrace/dynatrace-operator/test/secrets"
+	"github.com/Dynatrace/dynatrace-operator/test/setup"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const agentVersion = "VERSION"
+
+var injectionLabel = map[string]string{
+	"inject": "dynakube",
+}
+
+func AgentVersionSet(t *testing.T) features.Feature {
+	secretConfig, err := secrets.DefaultSingleTenant(afero.NewOsFs())
+	require.NoError(t, err)
+
+	versions := getAvailableVersions(secretConfig, t)
+	oldVersion, newVersion := assignVersions(t, versions, version.SemanticVersion{}, version.SemanticVersion{})
+
+	dk := dynakube.NewBuilder().
+		WithDefaultObjectMeta().
+		WithDynakubeNamespaceSelector().
+		ApiUrl(secretConfig.ApiUrl).
+		CloudNativeWithAgentVersion(&dynatracev1beta1.CloudNativeFullStackSpec{}, oldVersion).
+		Build()
+
+	agentVersionset := features.New("cloudnative with agent version set")
```

Comment:
nitpick: `specificAgentVersion` maybe is clearer, because when I read `agentVersionset` I thought for a second that its a set of agent versions 😅 

```suggestion
	specificAgentVersion := features.New("cloudnative with specific agent version set")
```